ratbag_emu/actuator.py

```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: MIT
+
+import logging
+import typing
+
+from abc import ABC, abstractmethod
+from typing import List
+
+if typing.TYPE_CHECKING:
+    from ratbag_emu.device import Device  # pragma: no cover
+
+
+class Actuator(ABC):
+    '''
+    Represents the firmware of the device
+
+    Transforms actions based on physical properties
+    '''
+    def __init__(self, owner: 'Device'):
+        self.__logger = logging.getLogger('ratbag-emu.actuator')
+
+        self._owner = owner
+
+        self._keys: List[str] = []
```

Comment:
... and use `typing.List` directly here.

It's never good to use the `from module import foo` syntax as we end up not knowing where each imported object comes from.

Reply:
Well, I think that's a great position, however I don't think it's necessary when importing type hint structures from `typing`. Having the full module path makes the code clunky and harder to read, especially when the types start getting more complex.

This is actually the standard way to do it, I believe. I had a long browse at python's source code I was not able to find any instance where they just imported the module.

Reply:
> Having the full module path makes the code clunky and harder to read, especially when the types start getting more complex.

Problem is here: I never used the `typing` module previously, so I will always wonder *where* this non standard keyword/class comes from.

> This is actually the standard way to do it, I believe. I had a long browse at python's source code I was not able to find any instance where they just imported the module.

This is not an argument: "everybody uses this, so this is the way it should be used".

Anyway. I am fine keeping the `from foo import blah` in this case, if this is something you truly believe will be clearer in the long run.

Reply:
> Problem is here: I never used the typing module previously, so I will always wonder where this non standard keyword/class comes from.

I understand.

> This is not an argument: "everybody uses this, so this is the way it should be used".

That was not really my point. I don't really care if other users use it this way, my point was that the Python developers do.

> Anyway. I am fine keeping the from foo import blah in this case, if this is something you truly believe will be clearer in the long run.

A more complex, but not outlandish, type definition like this

```
Union[List[int], Dict[str, int]]
```

would turn into this


```
typing.Union[typing.List[int], typing.Dict[str, int]]
```

which hurts readability a lot.